packages/remix/src/client/performance.tsx

```diff
@@ -1,46 +1,62 @@
-import { SEMANTIC_ATTRIBUTE_SENTRY_SOURCE } from '@sentry/core';
-import type { ErrorBoundaryProps } from '@sentry/react';
-import { WINDOW, withErrorBoundary } from '@sentry/react';
-import type { Transaction, TransactionContext } from '@sentry/types';
+import { SEMANTIC_ATTRIBUTE_SENTRY_SOURCE, getActiveSpan, getRootSpan } from '@sentry/core';
+import type { browserTracingIntegration as originalBrowserTracingIntegration } from '@sentry/react';
+import type { BrowserClient, ErrorBoundaryProps } from '@sentry/react';
+import {
+  WINDOW,
+  getClient,
+  startBrowserTracingNavigationSpan,
+  startBrowserTracingPageLoadSpan,
+  withErrorBoundary,
+} from '@sentry/react';
+import type { Span, Transaction, TransactionContext } from '@sentry/types';
 import { isNodeEnv, logger } from '@sentry/utils';
 import * as React from 'react';
 
 import { DEBUG_BUILD } from '../utils/debug-build';
 import { getFutureFlagsBrowser, readRemixVersionFromLoader } from '../utils/futureFlags';
 
-const DEFAULT_TAGS = {
-  'routing.instrumentation': 'remix-router',
-} as const;
-
-type Params<Key extends string = string> = {
+export type Params<Key extends string = string> = {
   readonly [key in Key]: string | undefined;
 };
 
-interface RouteMatch<ParamKey extends string = string> {
+export interface RouteMatch<ParamKey extends string = string> {
   params: Params<ParamKey>;
   pathname: string;
   id: string;
   handle: unknown;
 }
+export type UseEffect = (cb: () => void, deps: unknown[]) => void;
 
-type UseEffect = (cb: () => void, deps: unknown[]) => void;
-type UseLocation = () => {
+export type UseLocation = () => {
   pathname: string;
   search?: string;
   hash?: string;
   state?: unknown;
   key?: unknown;
 };
-type UseMatches = () => RouteMatch[] | null;
 
-let activeTransaction: Transaction | undefined;
+export type UseMatches = () => RouteMatch[] | null;
+
+export type RemixBrowserTracingIntegrationOptions = Partial<Parameters<typeof originalBrowserTracingIntegration>[0]> & {
+  useEffect?: UseEffect;
+  useLocation?: UseLocation;
+  useMatches?: UseMatches;
+  startTransactionOnPageLoad?: boolean;
+  startTransactionOnLocationChange?: boolean;
```

Comment:
we don't need this here and should instead rely on the new `instrumentXXX` options :)